moabb/pipelines/multi_pass.py

```diff
@@ -0,0 +1,111 @@
+from sklearn.base import BaseEstimator, TransformerMixin, clone
+from sklearn.feature_selection import SelectKBest, mutual_info_classif
+from pyriemann.spatialfilters import CSP
+from pyriemann.estimation import Covariances as Cov
+import numpy as np
+import logging
+
+log = logging.getLogger()
+
+
+class LogVariance(BaseEstimator, TransformerMixin):
+
+    def fit(self, X, y=None):
+        """fit."""
+        return self
+
+    def transform(self, X):
+        """transform, concatenating all frequency bands."""
+        assert X.ndim == 4
+        return np.concatenate([np.log(np.var(X[..., i], -1)) for i in range(X.shape[-1])], axis=1)
+
+
+class MultibandCovariances(BaseEstimator, TransformerMixin):
+
+    def __init__(self, estimator='scm', est=None):
+        """Init."""
+        if est is None:
+            self.est = Cov(estimator=estimator)
+        else:
+            self.est = est
+
+    def fit(self, X, y=None):
+        return self
+
+    def transform(self, X):
+        assert X.ndim == 4
+        assert X.shape[-1] != 1
+        out = np.stack([self.est.transform(X[..., i])
+                        for i in range(X.shape[-1])], axis=3)
+        return out
+
+
+class AverageCovariance(MultibandCovariances):
+
+    def transform(self, X):
+        return super().transform(X).mean(axis=-1)
+
+
+class FBCSP(BaseEstimator, TransformerMixin):
```

Comment:
cf my comment above, this can be done by :

```python
csp = make_pipeline(Covariance(), CSP(4), SelectKBest())
fbcsp = make_pipeline(FilterBank(csp), LDA())
```